mbus/MBus.py

```diff
@@ -158,4 +175,11 @@ def frame_data_xml(self, reply_data):
         if not xml_result:
             raise Exception("libmbus.mbus_frame_data_xml failed")
 
-        return xml_result.decode('ISO-8859-1')
+        ret_val = cast(xml_result, c_char_p).value.decode('ISO-8859-1')
+        self._libc.free(xml_result)
+
+        return ret_val
+
+    def frame_data_free(self, frame_data):
```

Comment:
It is actually called from the client code (the Genie MBus plugin in this case) once the structure is no longer needed.